app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt

```diff
@@ -1192,7 +1198,11 @@ class BrowserTabFragment : Fragment(), FindListener, CoroutineScope, DaxDialogLi
             .show()
     }
 
-    fun omnibarViews(): List<View> = listOf(clearTextButton, omnibarTextInput, privacyGradeButton, searchIcon)
+    fun omnibarViews(): List<View> = listOf(clearTextButton, omnibarTextInput, searchIcon)
+
+    override fun onAnimationFinished() {
```

Comment:
Agreed, I have made several changes due to this. Mainly because we were not using the renderer for the privacy grade so we weren't using `renderIfChanged` which would avoid this. The event would still be propagated but nothing would happen since the stated would be the same. Apart from that the animations also don't do anything since we check if they were previously running prior to cancelling them.

Reply:
"The event would still be propagated but nothing would happen". Yes, but my point is about why should I receive an event that didn't happen? I would expect that `onAnimationFinished` is called when animation was running and finishes.

Reply:
I've removed the logic from the animator helper and explicitly made the call to the viewmodel instead which checks if the state changed before propagating the state.